functests/common.go

```diff
@@ -57,3 +73,45 @@ func GetRandomPVC(storageClass string, quantity string) *k8sv1.PersistentVolumeC
 
 	return pvc
 }
+
+// GetDataValidatorJob creates random.img file and copies it to pvc's mountpoint /data
+func GetDataValidatorJob(pvc string) *k8sbatchv1.Job {
+	randomName := "test-job-" + rand.String(12)
+	job := &k8sbatchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: randomName,
+		},
+		Spec: k8sbatchv1.JobSpec{
+			Template: k8sv1.PodTemplateSpec{
+				Spec: k8sv1.PodSpec{
+					RestartPolicy: k8sv1.RestartPolicyNever,
+					Containers: []k8sv1.Container{
+						k8sv1.Container{
+							Name:  randomName,
+							Image: "busybox",
+							VolumeMounts: []k8sv1.VolumeMount{
+								k8sv1.VolumeMount{
+									MountPath: "/data",
+									Name:      "volume-to-debug",
+								},
+							},
+							Command: []string{"/bin/sh", "-c"},
+							Args:    []string{"dd if=/dev/zero of=/tmp/random.img bs=512 count=1;md5VAR1=$(md5sum /tmp/random.img | awk '{ print $1 }');cp /tmp/random.img /data/random.img;md5VAR2=$(md5sum /data/random.img | awk '{ print $1 }');if [[ \"$md5VAR1\" != \"$md5VAR2\" ]];then exit 1; fi"},
```

Comment:
Have you tested whether the `md5VAR=$(..)` expressions are evaluated before or after passing the string to `/bin/sh`?
If it is evaluated before, both those files won't exist and you'll get an empty md5sum for both `md5VAR1` and `md5VAR2` and the job will succeed regardless of outcome.

Reply:
Subshell ordering should guarantee that they are executed first, yes. My main concern, however, is that this is a little hard to read.

@xenolinux could you define the script in a variable earlier in this function, using `; \` to separate the commands with newlines, and providing a comment with a short paragraph describing what this script is trying to accomplish?

Reply:
Agreed with @jarrpa: this is a bit dense.

Reply:
Ack. I will work on it.

Reply:
@rohantmp Those are evaluated both times - before and after.

Reply:
@jarrpa Added new lines in the slice itself to make it readable with the comments saying what does the respective command do.